lib/earmark_parser/line_scanner.ex

```diff
@@ -120,48 +119,44 @@ defmodule EarmarkParser.LineScanner do
 
   defp _type_of(line, options = %Options{}, recursive) do
     {ial, stripped_line} = Helpers.extract_ial(line)
+    {content, indent} = count_indent(line, 0)
+    lt_four? = indent < 4
 
     cond do
-      match = Regex.run(~r/\A (\s*) \z/x, line) ->
-        [_, leading] = match
-        %Line.Blank{indent: String.length(leading), line: line}
+      content == "" ->
+        create_text(line, content, indent)
 
-      match = !recursive && Regex.run(~r/\A (\s{0,3}) <! (?: -- .*? -- \s* )+ > \z/x, line) ->
-        [_, leading] = match
-        %Line.HtmlComment{complete: true, indent: String.length(leading), line: line}
+      lt_four? && !recursive && Regex.run(~r/\A <! (?: -- .*? -- \s* )+ > \z/x, content) ->
```

Comment:
So the idea is instead of having a bunch of regexes checking for `(\s{0,3})`, we precompute it and store it on `lt_four?`.